ContactControles.js

```diff
@@ -0,0 +1,142 @@
+var app = angular.module('myApp', ["ngRoute"]);
+
+app.config(function($routeProvider,$locationProvider) {
+     $routeProvider.when('/', {
+               templateUrl : 'home.html',
+               controller: 'contactsCtrl'
+     }).when('/editContact/:contact_index', {
+               templateUrl : 'addContact.html',
+               controller: 'editCtrl'
+     }).when('/addContact', {
+               templateUrl : 'addContact.html',
+               controller: 'addCtrl'
+                         }).otherwise({
+         redirectTo: '/'
+                         });
+ });
+
+app.controller('contactsCtrl', function($scope, ContactService, $http, $location, $window) {
+  $scope.contacts = ContactService.getContacts(); //Yehia: Tabs Tabs
+  $scope.goToHome =function(path) {
+    $location.path(path);
+  }
+
+  $scope.goToEdit =function(contact_index) {
+    if(contact_index =="new") {
+      $location.url("/addContact");
+    }
+    else {
+      $location.url("/editContact/"+contact_index);
+    }
+  }
+
+  $scope.deleteContact = function(id) {
+    var deleteUser = $window.confirm('Are you sure you want to delete the Ad?');
+    if(deleteUser) {
+      ContactService.deleteContact(id);
+      alert("The Contact was deleted");
+    }
+    else  {
+      alert("The Contact wasn't deleted yet");
+    }
+  }
+
+});
+
+app.controller('editCtrl', function($scope,ContactService,$routeParams,$location){
+  $scope.edit=ContactService.edit;
+  var index = $routeParams.contact_index;
+  var contacts = ContactService.getContacts();
+  $scope.currentContact = contacts[index];
+    $scope.saveEditContact = function()
+    {
+    alert("Edit Successfully");
+    $location.path('/home');
+    }
+
+  $scope.goToHome =function(path){
+  $location.path(path);
+  }
+});
+
+app.controller('addCtrl', function($scope,ContactService,$routeParams,$location){
+$scope.edit = false;
+$scope.contacts= ContactService.getContacts();
+$scope.edit=true;

```

Comment:
`$scope.edit=true` flag is a bit confusing, since it's in the addCtrl and you are adding a new contact not editing it. :slightly_smiling_face: